hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDTableServiceClient.java

```diff
@@ -101,14 +101,15 @@ protected HoodieWriteMetadata<HoodieData<WriteStatus>> partialUpdateTableMetadat
       String instantTime,
       WriteOperationType writeOperationType) {
     if (isStreamingWriteToMetadataEnabled(table)) {
-      boolean enforceCoalesceWithRepartition = writeOperationType == WriteOperationType.CLUSTER && config.getBulkInsertSortMode() == BulkInsertSortMode.NONE;
+      boolean enforceCoalesceWithRepartition = writeOperationType == WriteOperationType.CLUSTER; // for other table services, enforceCoalesceWithRepartition will be false.
       if (enforceCoalesceWithRepartition) {
-        // check clustering plan for sort columns. only if there are no sort columns, then we might still set enforceCoalesceWithRepartition to true.
+        // check clustering plan for sort columns. only if there are no sort columns, then fallback to sort mode from write config.
         HoodieClusteringPlan clusteringPlan = ClusteringUtils.getClusteringPlan(
                 table.getMetaClient(), ClusteringUtils.getRequestedClusteringInstant(instantTime, table.getActiveTimeline(), table.getInstantGenerator()).get())
             .map(Pair::getRight).orElseThrow(() -> new HoodieClusteringException(
                 "Unable to read clustering plan for instant: " + instantTime));
-        enforceCoalesceWithRepartition = !clusteringPlan.getStrategy().getStrategyParams().containsKey(PLAN_STRATEGY_SORT_COLUMNS.key());
+        enforceCoalesceWithRepartition = (!clusteringPlan.getStrategy().getStrategyParams().containsKey(PLAN_STRATEGY_SORT_COLUMNS.key())
```

Comment:
Let's check if the value is non-null or empty instead of just checking the key is not present